### ISSUE TITLE
REPL with ARGV

### ISSUE THREAD

**Issue by ghost** [rakudo/rakudo#3050]
Python allows you to pass ARGV in interactive mode:

~~~
$ python3 - aa
>>> import sys
>>> sys.argv[1]
'aa'
~~~

~~~
$ python3 -- - aa
>>> import sys
>>> sys.argv[1]
'aa'
~~~

~~~
$ python3 -i - aa
>>> import sys
>>> sys.argv[1]
'aa'
~~~

as does PHP:

~~~
$ php -a -- aa
php > echo $argv[1];
aa
~~~

However Perl 6 seems to have no way to do this. These all hang:

~~~
perl6 - aa
perl6 -- - aa
perl6 --repl-mode=interactive - aa
perl6 --repl-mode=interactive -- - aa
~~~

and these:

~~~
perl6 -- aa
perl6 --repl-mode=interactive aa
perl6 --repl-mode=interactive -- aa
~~~

all error out:

~~~
Could not open aa. Failed to stat file: no such file or directory
~~~

**Comment by cup** [comment]
https://github.com/MoarVM/MoarVM/issues/1144

**Comment by lizmat** [comment]
Perhaps the `--repl-mode` argument should default to "interactive" **and** activate the REPL

**Comment by lizmat** [comment]
Fixed as of 2024.10-40-g238e313eb.  Will be available in the 2024.12 release:
```
% raku --repl-mode=tty 42 555
Welcome to Rakudo™ v2024.10.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2024.10.

To exit type 'exit' or '^D'
[0] > @*ARGS                                                                   
[42 555]
[1] > 
```
